AirSkyBoat/AirSkyBoat
# Bug fix: Wyvern (and maybe all other pets) continue to follow master while slept

**_I affirm:_**

## What does this pull request do?

There is a bug where wyverns continue to follow their master while slept. 

There are a couple ways this could be implemented but essentially using `CAIContainer::CanFollowPath` inside of `CPetController::DoRoamTick` is useful for checking to see if a pet can move at all. If it can't, return without doing anything. 

That said, I added the following code at `pet_controller.cpp:74` and it stopped the wyvern from following while slept. My assumption is that this behavior is expected for all pets.

```cpp
// if pet can't follow then don't
if (!PPet->PAI->CanFollowPath())
{
    return;
}
```

## Steps to test these changes

1. set job to DRG
2. open job abilities -> call wyvern
3. target wyvern
4. call `!exec target:addStatusEffect(xi.effect.SLEEP_I, 1, 0, 30)`
5. run away
6. wyvern should stay in same spot until sleep wears off and then roam to master

I uploaded a screen cap here: https://files.catbox.moe/3lpo0w.mp4
